### PR TITLE
S/pause resume projections

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/MultiUpstreamRowDownstream.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            }
+        }
+
+        @Override
+        public boolean setNextRow(Row row) {
+            return downstreamHandle == null || downstreamHandle.setNextRow(row);
+        }
+    };
+
+    public void downstreamHandleCallable(DownstreamHandleCallable downstreamHandleCallable) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/MultiUpstreamRowDownstream.java]: why is this method public?

File: sql/src/main/java/io/crate/operation/MultiUpstreamRowDownstream.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    public List<RowUpstream> upstreams() {
+        return registeredUpstreams;
+    }
+
+    public int pendingUpstreams() {
+        return pendingUpstreams.get();
+    }
+
+    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/MultiUpstreamRowDownstream.java]: this warning doesn't apply here?

File: sql/src/main/java/io/crate/operation/fetch/PositionalBucketMerger.java [hunk 3]
```diff
@@ -143,30 +140,42 @@ private void findLeastBucketIt() {
...
     }
 
+    @Override
+    public void resume() {
+        multiUpstreamRowUpstream.resume();
+    }
+
+    @Override
+    public boolean setNextRow(Row row) {
+        throw new UnsupportedOperationException("setNextRow not supported, use setNextBucket");
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/fetch/PositionalBucketMerger.java]: does the PositionalBucketMerger need to implement the Handle interface if it doesn't support it ?

File: sql/src/main/java/io/crate/operation/MultiUpstreamRowDownstream.java (hunk 1 above)
Comment (seut) [inline sql/src/main/java/io/crate/operation/MultiUpstreamRowDownstream.java]: how should the callable set otherwise? ctor?

**Comment by mfussenegger** [comment]
ack

**Pull request merged**